pyproject.toml

```diff
@@ -51,13 +56,13 @@ PyYAML = "^6.0"
 Jinja2 = "<3.1"
 pyfiglet = "^0.8.post1"
 lightgbm = "^3.3.2"
-FLAML = "^1.0.11"
+FLAML = {extras = ["automl"], version = "^2.3.3"}
 s3fs = ">=2022.8.2"
 sqlmodel = {version = "^0.0.19", optional = true}
 APScheduler = "^3.9.1"
 psycopg2-binary = {version = "^2.9.3", optional = true}
 segment-analytics-python = "^2.3.2"
-python-dotenv = "^0.21.0"
+python-dotenv = "^1.0"
 types-pyyaml = "^6.0.12.8"
```

Comment:
isn't this for type checking? you could put such dependencies in a lint subgroup. This does add some work - you would need to identify which deps you actually need to run the main package, and which ones you need for the tests, etc 
here's the relevant documentation: https://python-poetry.org/docs/managing-dependencies/#dependency-groups